test/unit/definitions/divorce/CaseTypeTab.test.js

```diff
@@ -1,11 +1,26 @@
 const expect = require('chai').expect;
 const assert = require('chai').assert;
 const { uniq, uniqWith, map, filter } = require('lodash');
+const { loadAllFiles, sortCaseTypeTabs } = require('../../utils/utils');
 
-const caseTypeTab = Object.assign(require('definitions/divorce/json/CaseTypeTab/CaseTypeTab'), {});
-const caseField = Object.assign(require('definitions/divorce/json/CaseField/CaseField'), {});
+const getAllCaseTypeTabDefinitions = loadAllFiles('CaseTypeTab');
+const caseTypeTab = getAllCaseTypeTabDefinitions(
```

Comment:
@madjava, shouldn't we have two sets of tests for these? One set for core+prod files and another set for core+nonProd files?
Otherwise, tests will pass but it won't really mean that either of these configuration sets will have all it needs to run isolated from the other.

Reply:
@jvictorcf You are making a point here. All the existing test all seem to combine all files together so followed same. Fancy bringing this up with with @yarpo on separating tests this way? My knowledge is still limited as to the impact doing it this way (if at all).

Reply:
Sure. We can do this as a separate thing. @yarpo, what do you think?

But, @madjava, while we don't play this, I'd only have the production files in your test (i.e. core + prod).
This way, we're protecting what's more precious (which is production)

Reply:
@jvictorcf Not sure how to play this suggestion. The new tab 'service application' would not exist in the core or prod one. its only in the deem-dispenced feature file. However the tests amongts other things also tests that the service application tab comes after the linked cases tab in that order (existing behaviour) and to do this it combines all existing tabs plus the new one, the . Any ideas?

Reply:
if you need to combine the prod file with the non-prod file, then you're testing something that doesn't exist in real life.
I think you only have a few valid choices to test here:

- Non-prod (core + non-prod files)
- Prod (core + prod files)
- Both
- None